scene/main/window.cpp

```diff
@@ -833,49 +859,34 @@ bool Window::is_visible() const {
 	return visible;
 }
 
-void Window::_update_window_size() {
-	// Force window to respect size limitations of rendering server.
-	RenderingServer *rendering_server = RenderingServer::get_singleton();
-	if (rendering_server) {
-		Size2i max_window_size = rendering_server->get_maximum_viewport_size();
+Size2i Window::_clamp_window_size(const Size2i &p_size) {
+	Size2i window_size_clamped = p_size;
+	Size2 minsize = get_clamped_minimum_size();
+	window_size_clamped = window_size_clamped.max(minsize);
 
-		if (max_window_size != Size2i()) {
-			size = size.min(max_window_size);
-			min_size = min_size.min(max_window_size);
-			max_size = max_size.min(max_window_size);
-		}
+	if (max_size_used != Size2i()) {
+		window_size_clamped = window_size_clamped.min(max_size_used);
 	}
 
-	Size2i size_limit;
-	if (wrap_controls) {
-		size_limit = get_contents_minimum_size();
-	}
+	return window_size_clamped;
+}
 
-	size_limit.x = MAX(size_limit.x, min_size.x);
-	size_limit.y = MAX(size_limit.y, min_size.y);
+void Window::_update_window_size() {
+	Size2i size_limit = get_clamped_minimum_size();
```

Comment:
Isn't this value cached in `Window::min_size`?

Reply:
This is the minimum window size can actually reach, it may be larger than `min_size` when `wrap_controls` is enabled.